Make tests runnable again, update MOGPFlags2, fix typos

Marlamin opened the pull request:
- Makes tests executable and target current .NET 8 LTS (tests cannot be a .NET Standard project).
- Renamed all occurences of "Entrys" typos to "Entries" **(note, possibly a breaking change to downstream projects)**.
- Rename `MOGPTerrainFlags` to `MOGPFlags2` and add newer flags.
- Some comment updates.
- Fix various typos.

No clue what the whitespace changes are about, I assume git rewrote them when I `git mv`'d them or VS did upon the namespace rename. There are no other changes in these files.

Luzifix commented:
Testing with [Kalimdor_25_55.zip](https://github.com/Luzifix/Warcraft.NET/files/3106992/Kalimdor_25_55.zip) failed. The MH2O Chunk look after save different.

Kaev commented:
The others LVF formats have still to be implemented. Currently LiquidVertexFormat 0 and 2 is supported. LiquidObject support needs DBC reading first.

Luzifix commented on Warcraft.NET/Files/ADT/Chunks/MH2O.cs:
```diff
@@ -0,0 +1,180 @@
+
+        public MH2O()
+        {
+
```
Remove whitespace

Luzifix commented on Warcraft.NET/Files/ADT/Chunks/MH2O.cs:
```diff
@@ -0,0 +1,180 @@
+        /// </summary>
+        public MH2OHeader[] MH2OHeaders { get; set; } = new MH2OHeader[256];
+
+        public MH2O()
```
Missing DocHead

Luzifix commented on Warcraft.NET/Files/ADT/Chunks/MH2O.cs:
```diff
@@ -0,0 +1,180 @@
+{
+    public class MH2O : IIFFChunk, IBinarySerializable
+    {
+        public const string Signature = "MH2O";
```
Missing DocHead

Luzifix commented on Warcraft.NET/Files/ADT/Chunks/MH2O.cs:
```diff
@@ -0,0 +1,180 @@
+
+namespace Warcraft.NET.Files.ADT.Chunks
+{
+    public class MH2O : IIFFChunk, IBinarySerializable
```
Missing DocHead

Luzifix commented on Warcraft.NET/Files/ADT/Chunks/MH2O.cs:
```diff
@@ -0,0 +1,180 @@
+        /// <inheritdoc/>
+        public uint GetSize()
+        {
+            int size = MH2OHeader.GetSize() * 256;
```
```suggestion
            return (uint)Serialize().Length;
```

Luzifix commented on Warcraft.NET/Files/ADT/Entrys/MH2OAttribute.cs:
```diff
@@ -0,0 +1,58 @@
+        /// Seems to be useable as visibility information
+        /// </summary>
+        public byte[] Fishable { get; set; } = new byte[8];
+        /// <summary>
```
Add missing whitespace

Luzifix commented on Warcraft.NET/Files/ADT/Entrys/MH2OHeader.cs:
```diff
@@ -0,0 +1,66 @@
+        /// Gets or sets the offset of the <see cref="MH2OInstance"/>s.
+        /// </summary>
+        public uint OffsetInstances { get; set; }
+        /// <summary>
```
Add whitespace

Luzifix commented on Warcraft.NET/Files/ADT/Entrys/MH2OHeader.cs:
```diff
@@ -0,0 +1,66 @@
+        /// <summary>
+        /// Gets or sets the amount of <see cref="MH2OInstance"/>.
+        /// </summary>
+        public uint LayerCount { get; set; }
```
Add whitespace

Luzifix commented on Warcraft.NET/Files/ADT/Entrys/MH2OHeader.cs:
```diff
@@ -0,0 +1,66 @@
+        /// <summary>
+        /// Gets or sets the offset of the <see cref="MH2OAttribute"/>s.
+        /// </summary>
+        public uint OffsetAttributes { get; set; }
```
Add whitespace

Luzifix commented on Warcraft.NET/Files/ADT/Entrys/MH2OHeader.cs:
```diff
@@ -0,0 +1,66 @@
+        /// <summary>
+        /// Gets or sets the <see cref="MH2OInstance"/>s of the current <see cref="MH2OHeader"/>.
+        /// </summary>
+        public MH2OInstance[] Instances { get; set; }
```
Add whitespace

Luzifix commented on Warcraft.NET/Files/ADT/Entrys/MH2OInstance.cs:
```diff
@@ -0,0 +1,109 @@
+        /// <summary>
+        /// Gets or sets the liquid type.
+        /// </summary>
+        public ushort LiquidTypeId { get; set; }
```
Add whitespace after any attribute

Luzifix commented on Warcraft.NET/Files/Structures/UVMapEntry.cs:
```diff
@@ -0,0 +1,8 @@
+﻿namespace Warcraft.NET.Files.Structures
```
Add comments see: https://github.com/Luzifix/Warcraft.NET/blob/master/Warcraft.NET/Files/Structures/Rotator.cs

Luzifix commented:
[CodeFactor](https://www.codefactor.io/repository/github/luzifix/warcraft.net) found multiple issues last seen at 971e836525e496ac7205772b3ee9a3d47834d242:

#### Insert parentheses within the arithmetic expression to declare the operator precedence.
[Warcraft.NET\Files\ADT\Chunks\MH2O.cs:50
](https://github.com/Luzifix/Warcraft.NET/blob/971e836525e496ac7205772b3ee9a3d47834d242/Warcraft.NET/Files/ADT/Chunks/MH2O.cs#L50)[Warcraft.NET\Files\ADT\Chunks\MH2O.cs:138
](https://github.com/Luzifix/Warcraft.NET/blob/971e836525e496ac7205772b3ee9a3d47834d242/Warcraft.NET/Files/ADT/Chunks/MH2O.cs#L138)[Warcraft.NET\Files\ADT\Chunks\MH2O.cs:91
](https://github.com/Luzifix/Warcraft.NET/blob/971e836525e496ac7205772b3ee9a3d47834d242/Warcraft.NET/Files/ADT/Chunks/MH2O.cs#L91)[Warcraft.NET\Files\ADT\Entrys\MH2OInstanceVertexData.cs:90
](https://github.com/Luzifix/Warcraft.NET/blob/971e836525e496ac7205772b3ee9a3d47834d242/Warcraft.NET/Files/ADT/Entrys/MH2OInstanceVertexData.cs#L90)[Warcraft.NET\Files\ADT\Entrys\MH2OInstance.cs:84
](https://github.com/Luzifix/Warcraft.NET/blob/971e836525e496ac7205772b3ee9a3d47834d242/Warcraft.NET/Files/ADT/Entrys/MH2OInstance.cs#L84)
#### A closing curly bracket must not be preceded by a blank line.
[Warcraft.NET\Files\ADT\Chunks\MH2O.cs:19
](https://github.com/Luzifix/Warcraft.NET/blob/971e836525e496ac7205772b3ee9a3d47834d242/Warcraft.NET/Files/ADT/Chunks/MH2O.cs#L19)
#### An opening curly bracket must not be followed by a blank line.
[Warcraft.NET\Files\ADT\Chunks\MH2O.cs:17
](https://github.com/Luzifix/Warcraft.NET/blob/971e836525e496ac7205772b3ee9a3d47834d242/Warcraft.NET/Files/ADT/Chunks/MH2O.cs#L17)
#### Complex Method
[Warcraft.NET\Files\ADT\Chunks\MH2O.cs:106-179
](https://github.com/Luzifix/Warcraft.NET/blob/971e836525e496ac7205772b3ee9a3d47834d242/Warcraft.NET/Files/ADT/Chunks/MH2O.cs#L106-L179)[Warcraft.NET\Files\ADT\Entrys\MH2OInstanceVertexData.cs:29-87
](https://github.com/Luzifix/Warcraft.NET/blob/971e836525e496ac7205772b3ee9a3d47834d242/Warcraft.NET/Files/ADT/Entrys/MH2OInstanceVertexData.cs#L29-L87)[Warcraft.NET\Files\ADT\Entrys\MH2OInstanceVertexData.cs:94-143
](https://github.com/Luzifix/Warcraft.NET/blob/971e836525e496ac7205772b3ee9a3d47834d242/Warcraft.NET/Files/ADT/Entrys/MH2OInstanceVertexData.cs#L94-L143)

Marlamin committed:
Rename typo'd directories

Marlamin committed:
Rename typo'd namespaces/variables

This is a breaking change for any libraries that use the incorrect namespace/variables references

Marlamin committed:
Target latest LTS for test project

Test projects need to be executable, they cannot be .NET Standard projects

Marlamin committed:
Rename/update MOGPFlags2

Marlamin committed:
Add notes for WMO shader 23

Marlamin committed:
Fix typos
